system/logaccess/ElasticStack/ElasticStackLogAccess.hpp

```diff
@@ -0,0 +1,122 @@
+    StringBuffer m_audienceIndexSearchPattern;
+
+    StringBuffer m_classSearchColName;
+    StringBuffer m_classIndexSearchPattern;
+
+    StringBuffer m_defaultDocType; //default doc type to query
+
+    elasticlient::Client m_esClient;
+    StringBuffer m_esConnectionStr;
+
+public:
+
+#ifdef _CONTAINERIZED
+    //In containerized world, most likely Elastic Search host is their default k8s hostname
+    static constexpr const char * DEFAULT_ES_HOST = "elasticsearch-master";
+#else
+    //In baremetal, localhost is good guess as any
+    static constexpr const char * DEFAULT_ES_HOST = "localhost";
+#endif
+
+    static constexpr const char * DEFAULT_ES_PROTOCOL = "http";
+    static constexpr const char * DEFAULT_ES_DOC_TYPE = "_doc";
+    static constexpr const char * DEFAULT_ES_PORT = "9200";
+
+    static constexpr int DEFAULT_ES_DOC_LIMIT = 100;
+    static constexpr int DEFAULT_ES_DOC_START = 0;
+
+    static constexpr const char * DEFAULT_TS_NAME = "@timestamp";
+    static constexpr const char * DEFAULT_INDEX_PATTERN = "filebeat*";
+    static constexpr const char * DEFAULT_LOG_COLUMN_NAME = "message";
+
+    static constexpr const char * DEFAULT_HPCC_LOG_SEQ_COL = "hpcc.log.sequence";
+    static constexpr const char * DEFAULT_HPCC_LOG_TIMESTAMP_COL = "hpcc.log.timestamp";
+    static constexpr const char * DEFAULT_HPCC_LOG_MESS_COL = "hpcc.log.message";
+    static constexpr const char * DEFAULT_HPCC_LOG_AUD_COL = "hpcc.log.audience";
+    static constexpr const char * DEFAULT_HPCC_LOG_PROCID_COL = "hpcc.log.procid";
+    static constexpr const char * DEFAULT_HPCC_LOG_THREADID_COL = "hpcc.log.threadid";
+    static constexpr const char * DEFAULT_HPCC_LOG_TYPE_COL = "hpcc.log.class";
+    static constexpr const char * DEFAULT_HPCC_LOG_MESSAGE_COL = "hpcc.log.message";
+    static constexpr const char * DEFAULT_HPCC_LOG_JOBID_COL = "hpcc.log.jobid";
+    static constexpr const char * DEFAULT_HPCC_LOG_COMPONENT_COL = "kubernetes.container.name";
+
+    static constexpr const char * LOGMAP_INDEXPATTERN_PATH = "logmap/global/@storename";
+    static constexpr const char * LOGMAP_COLUMNNAME_PATH = "logmap/global/@searchcolumn";
+    static constexpr const char * LOGMAP_TIMESTAMPCOL_PATH = "logmap/global/@timestampcolumn";
+
+    ElasticStackLogAccess(const std::vector<std::string> &hostUrlList, IPropertyTree & logAccessPluginConfig);
+    virtual ~ElasticStackLogAccess() override = default;
+
+    // IRemoteLogAccess methods
+    virtual bool fetchLog(const LogAccessConditions & options, StringBuffer & returnbuf, LogAccessLogFormat format) override;
+    virtual const char * getRemoteLogAccessType() const override { return type; }
+    virtual IPropertyTree * queryLogMap() const override { return m_pluginCfg->queryPropTree("logmap");}
+    virtual const char * fetchConnectionStr() const override { return m_esConnectionStr.str();}
+
+    // Elastic Stack specific
+    cpr::Response performESQuery(const LogAccessConditions & options);
```

Comment:
This looks like a private method - why would anything outside the class call this?

Reply:
These functions are useful for anybody using this library independently of the Ilogaccess interface making them private ensures they are not used (sans further changes). How much improvement do you see by forcing the library to be completely coupled to the Ilogaccess interface? @ghalliday